store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9064,6 +9071,97 @@ public Mountpoint refreshMountpoint(OperationContext octxt, int mountpointId, St
         return getMountpointById(octxt, mountpointId);
     }
 
+    public void createFileSharedWithMe(Mailbox mbox, OperationContext octxt, int folderId, String fileName,
+            String ownerAccountId, int ownerFileId, String remoteUuid, String fileOwnerName,
+            String contentType, long fileSize, String rights) throws ServiceException {
+
+        CreateFileSharedWithMe redoRecorder = new CreateFileSharedWithMe(mId, folderId, fileName, ownerAccountId,
+                ownerFileId, remoteUuid, fileOwnerName, contentType, rights);
+
+        boolean success = false;
+        try {
+            beginTransaction("createFileSharedWithMe", octxt, redoRecorder);
+            CreateFileSharedWithMe redoPlayer = (CreateFileSharedWithMe) currentChange().getRedoPlayer();
+
+            int mptId = getNextItemId(redoPlayer == null ? ID_AUTO_INCREMENT : redoPlayer.getId());
+            String uuid = redoPlayer == null ? UUIDUtil.generateUUID() : redoPlayer.getUuid();
+            UnderlyingData data = new UnderlyingData();
+            data.uuid = remoteUuid + ":" + ownerFileId;
+            data.id = mptId;
+            data.type = Type.DOCUMENT.toByte();
+            data.folderId = folderId;
+            data.date = mbox.getOperationTimestamp();
+            data.name = fileName;
+            data.setSubject(fileName);
+            data.size = fileSize;
+            data.metadata = encodeFileSharedMetadata(new Metadata(), fileOwnerName, ownerAccountId,
+                    ownerFileId, contentType, rights).toString();
+            data.contentChanged(mbox);
+            new DbMailItem(mbox).create(data);
+            Document doc = new Document(mbox, data);
+            doc.finishCreation(null);
+            success = true;
+        } finally {
+            endTransaction(success);
+        }
+    }
+
+    public void deleteFileSharedWithMe(Mailbox mbox, OperationContext octxt, String remoteUuid, int ownerFileId,
+            String ownerAccountId, String granteeAccountID) throws ServiceException {
+        CreateFileSharedWithMe redoRecorder = new CreateFileSharedWithMe(mId, remoteUuid);
+        boolean success = false;
+        try {
+            beginTransaction("deleteFileSharedWithMe", octxt, redoRecorder);
+            Document item = (Document) mbox.getItemByUuid(remoteUuid + ":" + ownerFileId, Type.DOCUMENT);
+            if (item.getOwnerAccountId().equals(ownerAccountId) && item.getAccountId().equals(granteeAccountID)) {
+                DbMailItem.delete(mbox, Collections.singletonList(item.getId()), false);
+                mbox.getFolderById(ID_FOLDER_FILE_SHARED_WITH_ME).updateSize(-1, -1, -item.getSize()); //update the folder item count and size accordingly
```

Comment:
for comments add space after `//` and move the comment above the line

Reply:
moved the comment above line and formatted accordingly